pipeline/load.py

```diff
@@ -0,0 +1,331 @@
+
+            if count == 0:
+                cur.execute("""
+                    INSERT INTO delta.Scientific_Names (scientific_name)
+                    VALUES (%s);
+                """, (scientific_name))
+                print(f"Inserted: {scientific_name}")
+            else:
+                print(f"Duplicate: {scientific_name}")
+
+
+def insert_location(conn: 'connection', plant_df: pd.DataFrame) -> None:
+    """Inserts the location of a plant into the database
+        Ignores duplicate entries"""
+    with conn.cursor() as cur:
+        for index, row in plant_df.iterrows():
+            latitude = row["Latitude"]
+            longitude = row["Longitude"]
+            town = row["Town"]
+            country_code = row["Country_Code"]
+            continent = row["Continent"]
+            city = row["City"]
+
+            cur.execute("""SELECT continent_id
+                        FROM delta.Continents
+                        WHERE continent_name = %s;
+                        """, (continent))
+            continent_id = cur.fetchone()[0]
+
+            cur.execute("""
+                SELECT COUNT(*)
+                FROM delta.Locations
+                WHERE latitude = %s AND longitude = %s AND town = %s
+                        AND country_code = %s AND continent_id = %s AND city = %s;
+            """, (latitude, longitude, town, country_code, continent_id, city))
+
+            count = cur.fetchone()[0]
+
+            if count == 0:
+                cur.execute("""
+                    INSERT INTO delta.Locations (latitude, longitude, town, country_code, continent_id, city)
+                    VALUES (%s, %s, %s, %s, %s, %s);
+                """, (latitude, longitude, town, country_code, continent_id, city))
+                print(
+                    f"Inserted: ({latitude}, {longitude}, {town}, {country_code}, {continent_id}, {city})")
+            else:
+                print(f"""Duplicate: {latitude}, {longitude}, {
+                      town}, {country_code}, {continent_id}, {city}""")
+
+
+def insert_plants(conn: 'connection', plant_df: pd.DataFrame) -> None:
+    """Inserts plant information into the database
+        Ignores duplicate entries"""
+    with conn.cursor() as cur:
+        for index, row in plant_df.iterrows():
+            plant_id = row["plant_id"]
+            plant_name = row["name"]
+            scientific_name = row["scientific_name"]
+            latitude = row["Latitude"]
+            longitude = row["Longitude"]
+            image_url = row["images.original_url"]
+            town = row["Town"]
+            country_code = row["Country_Code"]
+            continent = row["Continent"]
+            city = row["City"]
+
+            try:
+                cur.execute("""SELECT continent_id
+                            FROM delta.Continents
+                            WHERE continent_name = %s;
+                            """, (continent))
+                continent_id = cur.fetchone()[0]
```

Comment:
We have a lot of separate executes in this program. Instead, we can have a generalised function that takes in query (table and conditions) and values to extract repetitive logic into a reusable function.